ui/pages/permissions-connect/snaps/snaps-connect/snaps-connect.js

```diff
@@ -57,7 +57,11 @@ export default function SnapsConnect({
   const snaps = getDedupedSnaps(request, currentPermissions);
 
   const SnapsConnectContent = () => {
-    const { hostname: trimmedOrigin } = useOriginMetadata(origin) || {};
+    let trimmedOrigin = (useOriginMetadata(origin) || {})?.name;
+
+    if (origin?.startsWith('local:') || origin?.startsWith('npm:')) {
```

Comment:
Feels like we should have a utility function for this in `snaps-utils`, Maarten was adding one here: https://github.com/MetaMask/snaps/pull/2634/files#diff-ef92bcefb67c52b36ad3e20af33cd46bcbd08d2ab94fd2bc6421ea2838634366R317-R319

Reply:
I can duplicate that function here in the extension and we can replace it once we get that into a release

Reply:
Makes sense 👍

Reply:
Done: [7aef11b](https://github.com/MetaMask/metamask-extension/pull/26715/commits/7aef11b5b8a61a3dd5f96a37ee9071409f739c05)